### PR TITLE
Staging to master

### PR DESCRIPTION
**Pull request by jarenyach**

### CODE REVIEW

**Comment by jarenyach** [comment]
@dcarroll This is the same details from the sfdx-ci pull request.  The readme and enc key has been updated.

**Review by dcarroll — APPROVED** [review]

**Review by bbuffing — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#Gives permission to install SFDX
+sudo: true
+
+before_install:
+#
+# The before_install step will download the SFDX CLI and install it
```
Comment (ekapner) [inline .travis.yml]: The before_install steps downloads Salesforce CLI and installs it

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#Assign variables
+- export CLIURL=https://developer.salesforce.com/media/salesforce-cli/sfdx-linux-amd64.tar.xz  
+- export SFDX_AUTOUPDATE_DISABLE=false
+- export SFDX_USE_GENERIC_UNIX_KEYCHAIN=true
+- export SFDX_DOMAIN_RETRY=300
+- export SFDX_DISABLE_APP_HUB=true
+- export SFDX_LOG_LEVEL=DEBUG
+#Create SFDX directory
+- mkdir sfdx
+#Install SFDX
```
Comment (ekapner) [inline .travis.yml]: @clairebianchi - what specifically are we installing here?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- wget -qO- $CLIURL | tar xJ -C sfdx --strip-components 1
+- "./sfdx/install"
+- export PATH=./sfdx/$(pwd):$PATH
+
+before_script:
+#
+# The before_script step initializes varibles used in the script step and authenticates to the devhub using JWT grant
```
Comment (ekapner) [inline .travis.yml]: The before_script step initializes variables used in the script step and authenticates to the Dev Hub using JWT-based flow

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+#
+
+#Assign variables
+- export ROOTDIR=force-app/main/default/
+- export TESTLEVEL=RunLocalTests
+#Be sure to update the package ID (Package Id will start with 0Ho) or package alias here from the sfdx-project.json file
```
Comment (ekapner) [inline .travis.yml]: (package ID starts with OHo)

File: .travis.yml (hunk 2 above)
Comment (jarenyach) [reply .travis.yml]: The Salesforce CLI is being installed.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+- export PACKAGENAME="0Ho0N000000CaT3SAK"
+- export PACKAGEVERSION=""
+#outputs CLI version
+- sfdx --version
+#outputs salsforcedx plugin version
+- sfdx plugins --core
+#Authorize DevHub
```
Comment (ekapner) [inline .travis.yml]: Authorize Dev Hub

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+- sfdx force:auth:jwt:grant --clientid $CONSUMERKEY --jwtkeyfile assets/server.key --username $USERNAME --setdefaultdevhubusername --setalias HubOrg
+
+script:
+#Create Scratch Org
```
Comment (ekapner) [inline .travis.yml]: Create scratch org

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- sfdx force:org:create --targetdevhubusername HubOrg --setdefaultusername --definitionfile config/project-scratch-def.json --setalias ciorg --wait 10 --durationdays 1
+- sfdx force:org:display --targetusername ciorg
+#Push Source Scratch Org
```
Comment (ekapner) [inline .travis.yml]: Push source to scratch org

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+- sfdx force:source:push --targetusername ciorg
+#Run Unit Tests on Scratch Org
```
Comment (ekapner) [inline .travis.yml]: Run unit tests on scratch org

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+- sfdx force:apex:test:run --targetusername ciorg --wait 10 --resultformat tap --codecoverage --testlevel $TESTLEVEL
+#Delete Scratch Org
```
Comment (ekapner) [inline .travis.yml]: "scratch org" should be lowercase - assume you'll fix from this point forward

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+- sfdx force:org:delete --targetusername ciorg --noprompt
+#Create Package Version
```
Comment (ekapner) [inline .travis.yml]: Same with "package version" - should be lowercase

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+- PACKAGEVERSION="$(sfdx force:package:version:create --package $PACKAGENAME --installationkeybypass --wait 10 --json --targetdevhubusername HubOrg | jq '.result.SubscriberPackageVersionId' | tr -d '"')"
+- echo "Package Version Id - "${PACKAGEVERSION}
+- echo "Waiting for 5 minutes for package replication"
+- sleep 5m
+#Create Scratch Org
+- sfdx force:org:create --targetdevhubusername HubOrg --setdefaultusername --definitionfile config/project-scratch-def.json --setalias installorg --wait 10 --durationdays 1
+- sfdx force:org:display --targetusername installorg
+#Install package in Scratch Org
+- sfdx force:package:install --package $PACKAGEVERSION --wait 10 --targetusername installorg
+#Run Unit Tests on Scratch Org
```
Comment (ekapner) [inline .travis.yml]: Same for "unit tests"

File: .travis.yml (hunk 2 above)
Comment (jarenyach) [reply .travis.yml]: I updated the comment to call out what is getting installed.

**Pull request merged**